### ISSUE TITLE
Realms are not updating

### ISSUE THREAD

**Issue by b-morgan** [Auctionator/auctionator-app-bugs#15]
**Describe the bug**
Realms are not updating. I have closed and opened Auctionator Pro multiple times including a complete system restart and my realm data shows last downloaded 3 days ago

**Expected behavior**
The claim is updates every hour so I expect when the application is running, that the last downloaded times would be less than 1 hour old.

**Screenshots**
![image](https://github.com/user-attachments/assets/8c986f2c-817f-4a8e-96d3-e44f379f5c01)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version 27

**Additional context**

**Logs**

[main.log](https://github.com/user-attachments/files/16702958/main.log)

**Comment by b-morgan** [comment]
Apparently, the realms only update at the top of the hour. They do not update when the application is started.

**Comment by b-morgan** [comment]
At 7am, the realms last downloaded times updated. Please explain the algorithm and what detection techniques are being used to decide when to update the realms and when to check for updates to the application.

**Comment by jrob8577** [comment]
The schedule _should_ be:
1. On startup (this may be delayed a few seconds as the app discovers subscription info)
2. On the hour

If the computer goes into power saver mode, the download schedules are paused.

**Comment by jrob8577** [comment]
I see the download failure in the log; that is attributable to subscription discovery (and will be deferred and retried until subscription information is available in the next app release). Does your computer sleep overnight? 

I'll also do some research and see if there are other events that cause the application to be paused.

**Comment by b-morgan** [comment]
> Does your computer sleep overnight?

No, it does not. Backups and other tasks keep it awake. There is a screensaver that turns the monitor(s) off.

**Comment by b-morgan** [comment]
If it is the screensaver that pauses the updates, then if an update occurs when the screensaver ceases (not sure what to call the event when it wakes up), I'm OK with that. I.E. I don't need hourly updates when I'm not at the computer, but when I am at the computer, I would prefer not to wait until the top of the hour to get an update.

**Comment by jrob8577** [comment]
I'll look in to the possibility of the screen saver; that explanation sounds correct to me from observed behavior while testing on my end, but will verify. Thanks for the detailed info!
